app/controllers/tiny_mce_assets_controller.rb

```diff
@@ -1,6 +1,11 @@
 # frozen_string_literal: true
 
 class TinyMceAssetsController < ApplicationController
+  before_action :load_vars, only: %i(marvinjs_show marvinjs_update)
+
+  before_action :check_read_permission, only: %i(marvinjs_show marvinjs_update)
+  before_action :check_edit_permission, only: %i(marvinjs_update)
+
```

Comment:
There are no permission checking for download and create?

Reply:
For create we can't check permission because here same issue how for TinyMce images. It is not connected to any object.